src/hotspot/share/opto/runtime.cpp

```diff
@@ -1377,7 +1377,7 @@ address OptoRuntime::handle_exception_C(JavaThread* current) {
   // deoptimized frame
 
   if (nm != NULL) {
-    RegisterMap map(current, false /* update_map */, false /* process_frames */);
+    RegisterMap map(current, false);
```

Comment:
Should you add `MACOS_AARCH64_ONLY(ThreadWXEnable wx(WXWrite, current));` here, as you suggested in CR?

Reply:
I reverted back to exactly what we used to do here prior to JDK-8265702, to avoid affecting anything but ZGC. I'm placing the ThreadWXEnable in the ZGC nmethod barrier since we want these as close to the patching code as possible, and we can end up in there through multiple paths.

Reply:
I see. Thanks.